vault/cors.go

```diff
@@ -72,19 +84,29 @@ func (c *Core) loadCORSConfig() error {
 
 // Enable takes either a '*' or a comma-seprated list of URLs that can make
 // cross-origin requests to Vault.
-func (c *CORSConfig) Enable(urls []string) error {
+func (c *CORSConfig) Enable(urls []string, headers []string) error {
 	if len(urls) == 0 {
-		return errors.New("the list of allowed origins cannot be empty")
-	}
-
-	if strutil.StrListContains(urls, "*") && len(urls) > 1 {
+		urls = append(urls, "*")
+	} else if strutil.StrListContains(urls, "*") {
 		return errors.New("to allow all origins the '*' must be the only value for allowed_origins")
 	}
 
 	c.Lock()
 	c.AllowedOrigins = urls
 	c.Unlock()
 
+	c.Lock()
+	if len(c.AllowedHeaders) == 0 {
+		c.AllowedHeaders = append(c.AllowedHeaders, stdAllowedHeaders...)
+	}
+
+	// Allow the user to add additional headers to the list of
+	// headers allowed on cross-origin requests.
+	if len(headers) > 0 {
+		c.AllowedHeaders = append(c.AllowedHeaders, headers...)
```

Comment:
Enable certainly _could_ be called multiple times, because this code allows new headers to be appended to the existing set. This workflow, however, clashes with the workflow for setting origins. Your suggestion is correct.